gerritlab/merge_request.py

```diff
@@ -168,14 +168,13 @@ def refresh(self):
         r.raise_for_status()
         self._set_data(r.json())
         
-    def wait_until_stable(self):
+    def wait_until_stable(self, commit):
         """
-        Poll GitLab until the changes_count field has a value of "1".
-        Other values indicate that GitLab hasn't reacted to a push yet.
+        Poll the MR until the "sha" field matches that of `commit`.
         """
         while True:
             self.refresh()
-            if self._changes_count == "1":
+            if self._sha == commit.commit.hexsha:
```

Comment:
Is `self._sha` the SHA of the latest commit in the MR? Not sure how GitLab gives an MR a SHA.

`self.get_commits()` can be used to return the commits in the MR if this doesn't work.

Reply:
The doc says: `Diff head SHA of the merge request.`  The doc for `diff_refs` says `References of the base SHA, the head SHA, and the start SHA for this merge request. Corresponds to the latest diff version of the merge request. Empty when the merge request is created, and populates asynchronously.`

Will this be enough? I'm not sure. I'd like to not make an additional API call per MR if it is avoidable.  I'm hoping your stress test will exercise it enough to build confidence.

Reply:
Thanks! I just did the stress test and they all passed except for a failure that was caused by two tests creating the same branch (i.e. the same master-xxxx branch name. Maybe we need more characters in the branch names 🙂). So this PR should be enough. Let's check in this.

Reply:
I had a pending todo item to request extending the number of characters used (up to the entire Change-Id which would have parity with Gerrit), possibly making it configurable.  For Wikimedia Foundation projects there could be a lot of people using this tool, all targeting the same branch so the chance of collision is nontrivial.

Reply:
Sounds great!

Reply:
Filed as #39.